### PR TITLE
design for setting boot mode explicitly

### PR DESCRIPTION
**Pull request by dhellmann**
also mark implicit-boot-mode as implementable

### CODE REVIEW

**Comment by stbenjam** [comment]
This looks good to me :+1: I like the combination of implicit defaults, with the ability to override.

**Comment by dhellmann** [comment]
@pierrecregut you may have input on this design

**Comment by maelk** [comment]
looks good to me!

**Comment by stbenjam** [comment]
Oh I can't lgtm this one anyway :)

/hold cancel

File: design/baremetal-operator/explicit-boot-mode.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+### Non-Goals
+
+- Change the implicit boot mode selection design.
+
+## Proposal
+
+Add a new optional API input field, `spec.bootMode`, with possible
+values `UEFI` or `legacy`. If no value is provided, the value from the
+BMC method will be used.
```
Comment (juliakreger) [inline design/baremetal-operator/explicit-boot-mode.md:42]: Maybe "the system running default will be used."

Which would be the truth, the BMC doesn't entirely govern system configuration, it just provides a window and set of knobs.

Comment (dhellmann) [reply design/baremetal-operator/explicit-boot-mode.md:42]: Here I'm talking about the method on the BMC type, not the actual hardware management controller.

**Pull request merged**